src/solver/expressions/include/antares/solver/expressions/visitors/LinearStatus.h

```diff
@@ -20,121 +20,115 @@
 */
 #pragma once
 
+#include <ostream>
+
 namespace Antares::Solver::Visitors
 {
 
-class LinearStatus
+enum class LinearStatus : char
 {
-public:
-    enum Status
-    {
-        CONSTANT,
-        LINEAR,
-        NON_LINEAR
-    };
-
-    constexpr LinearStatus() = default;
-
-    constexpr LinearStatus(const Status& status):
-        status_(status)
-    {
-    }
-
-    constexpr LinearStatus(const LinearStatus& other) = default;
-
-    constexpr LinearStatus operator*(const LinearStatus& other)
-    {
-        switch (other)
-        {
-        case LinearStatus::NON_LINEAR:
-            return LinearStatus::NON_LINEAR;
-        case LinearStatus::CONSTANT:
-            return *this;
-        case LinearStatus::LINEAR:
-            if (status_ == LinearStatus::CONSTANT)
-            {
-                return other;
-            }
-            else
-            {
-                return LinearStatus::NON_LINEAR;
-            }
-        default:
-            return LinearStatus::NON_LINEAR;
-        }
-    }
-
-    constexpr LinearStatus operator/(const LinearStatus& other)
-    {
-        switch (other)
-        {
-        case LinearStatus::NON_LINEAR:
-        case LinearStatus::LINEAR:
-            return LinearStatus::NON_LINEAR;
-        case LinearStatus::CONSTANT:
-            return *this;
-        default:
-            return LinearStatus::NON_LINEAR;
-        }
-    }
+    CONSTANT = 0,
+    LINEAR = 1,
+    NON_LINEAR = 2
+};
 
-    constexpr LinearStatus operator+(const LinearStatus& other)
-    {
-        switch (other)
-        {
-        case LinearStatus::NON_LINEAR:
-            return LinearStatus::NON_LINEAR;
-        case LinearStatus::CONSTANT:
-            return *this;
-        case LinearStatus::LINEAR:
-            if (other == LinearStatus::CONSTANT || other == LinearStatus::LINEAR)
-            {
-                return other;
-            }
-            else
-            {
-                return LinearStatus::NON_LINEAR;
-            }
-        default:
-            return LinearStatus::NON_LINEAR;
-        }
-    }
+constexpr char pair(LinearStatus a, LinearStatus b)
```

Comment:
What's the problem with using classic **std::pair** ?
Why do we need the help of **char** ?

Reply:
`switch..case` only accepts integers (int, unsigned int, char, etc.), std::pair is not an integer or convertible to an integer

Reply:
ok, then why do we use switch.. case, and not a std::map that associates std::pair of enum value to enum value.
Is there a problem with this solution ?

Reply:
Yes, it's not `constexpr`.